youtube_dl/extractor/odnoklassniki.py

```diff
@@ -137,7 +137,7 @@ def _real_extract(self, url):
         video_id = self._match_id(url)
 
         webpage = self._download_webpage(
-            'http://ok.ru/video/%s' % video_id, video_id)
+            'https://ok.ru/video/%s' % video_id, video_id)
```

Comment:
Use scheme of the input URL.

Reply:
The original extractor did not do this. In all the tests, the request made is always __http__://ok.ru/video/%s' % video_id, video_id). Since the original extractor always used the same request, and always hit a redirect to https://ok.ru/... I would suggest that blocking this simple fix seems undesirable since the outcome is an extractor that will continue to leak unencrypted http. If you could lay out more detailed reasons why the simple fix to eliminate unencrypted http wouldn't be better than requiring something totally different that has not been the case throughout the history of the extractor, and perhaps how you would prefer it to be in more detail, maybe I can try to satisfy, but otherwise I don't see the logic of blocking this fix that should have been implemented years ago.

Reply:
Details on what? That's exactly how browser will work - if one requests to open http URL then http URL must be opened following by a redirect to https (if any). youtube-dl intention is to mimic browser's behavior in the first place, not introducing some levels of protection from leaking or whatsoever.

Reply:
That is __not__ how the extractor works in its present form... The extractor currently forces you to go unencrypted no matter what the input (whether https://ok.ru, http://m.ok.ru or http://mobile.ok.ru etc) Check the examples yourself… Hence it seems wrong to block my PR for something that the merged extractor does not do anyway.

```
youtube-dl --print-traffic http://m.ok.ru/video/20079905452
[Odnoklassniki] 20079905452: Downloading webpage
send: b'GET /video/20079905452 HTTP/1.1\r\nHost: ok.ru\r\nUser-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/71.0.3578.22 Safari/537.36\r\nAccept-Charset: ISO-8859-1,utf-8;q=0.7,*;q=0.7\r\nAccept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8\r\nAccept-Encoding: gzip, deflate\r\nAccept-Language: en-us,en;q=0.5\r\nConnection: close\r\n\r\n'
reply: 'HTTP/1.1 301 Moved Permanently\r\n'
header: Server: Apache
header: Date: Mon, 04 Jan 2021 00:01:20 GMT
header: Content-Length: 0
header: Connection: close
header: Location: https://ok.ru/video/20079905452 
``` 

or

```
youtube-dl --print-traffic http://mobile.ok.ru/video/20079905452
[Odnoklassniki] 20079905452: Downloading webpage
send: b'GET /video/20079905452 HTTP/1.1\r\nHost: ok.ru\r\nUser-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3761.2 Safari/537.36\r\nAccept-Charset: ISO-8859-1,utf-8;q=0.7,*;q=0.7\r\nAccept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8\r\nAccept-Encoding: gzip, deflate\r\nAccept-Language: en-us,en;q=0.5\r\nConnection: close\r\n\r\n'
reply: 'HTTP/1.1 301 Moved Permanently\r\n'
header: Server: Apache
header: Date: Mon, 04 Jan 2021 00:01:20 GMT
header: Content-Length: 0
header: Connection: close
header: Location: https://ok.ru/video/20079905452
``` 

The extractor in its current repo form already does not match input (which makes me wonder why it wasn't held up like this when it was merged??). It does not make any requests to m.ok.ru or mobile.ok.ru, it just takes the ID 20079905452 and then requests http://ok.ru/video/20079905452

Your decision to obstruct (which I feel many other PR submitters would agree you seem to have precedent for doing indiscriminately) is to leave this extractor in a state that exactly does __not__ meet your requirements of matching input...it forces HTTP no matter what you input. 

It is a very basic best practice to use TLS when possible, and browsers increasingly are moving toward zero support for unencrypted HTTP so this would become irrelevant and seem a sensible thing to do and in fact is something many current extractors already do.

Now, if you want to make your own commit after this one to change it to match input, you can of course do so. 

However, the pull I have put forward does not reduce the extractor's current level of emulation of browser behavior whatsoever, and simply makes the extractor a tiny bit faster and safer. The issue that you say blocks it from being merged seems strange given the extractor code already does not meet the requirements you say block the PR.

Reply:
Do you even read what's addressed to you? If one explicitly wants to access via http you **must not** force it to https. Current code uses http because at the time of writing such URLs were not available via https at all. I'm not going to argue further - your either do what's requested or this will be rejected.

Reply:
Do you even know what professional courtesy is or democratic coding principles are? Discussion is not "argu[ing]". There is zero logic in requiring this. You say "must not force it to https" but give no real reason why. Okay, to mimic a browser, and to allow you to "explicitly access via http" (even though you won't get anything but a redirect and the rest of the requests will all use TLS so you get nothing useful but a short term leak via a legacy protocol and a redirect) so there is no technical argument in favor of using http at all.

It is nonsensical to say the current code does explicitly use http because it was the only available option when added (note I remember seeing this very same issue back in 2016 or 2017 and OK.RU had TLS then and that is a while ago now...) because you specifically mandate the PR should make the extractor "mimic a browser". If that is your attitude, then the extractor should have been built from the start to simply use the input URL (again it doesn't do that from a domain perspective either - do you actually read anything?, it just talks to http://ok.ru/video/ID) and if someone used __https__://ok.ru/video/ID, the extractor should have tried, and simply failed or errored since ok.ru would not respond properly on port 443, not just use the unencrypted URL automatically. If I open a browser and try to open an https version of a broken old site that does not support https, the browser stalls or gives an error, it does not just open the unencrypted page. Your logic is really funny.

"your either do what's requested or this will be rejected." what an ego you have...you might try to learn English a bit more before being so pedantic.

I can't wait to see how youtube-dl will fair when the web pretty much turns off HTTP and QUIC becomes the norm - I look forward to many more hilarious/pityful, bureaucratic reasons given for closing, locking and dismissing issues and PRs - you (and anyone else) might want to look at how many PRs you add the pending label to that get closed and/or locked  because the submitter got fed up with your attitude to anyone trying to fix your mess of broken extractors - maybe you should just delete all of those extractors already so people realize the situation and fork.

I am closing this PR now, and it can settle in the dust with the rest of the pile of closed/locked good faith attempts by people who used to believe in this project to improve things that were met with nothing but arrogance and incompetence. Good luck, you will need it eventually if this continues to be your attitude to people who care about your project, and more generally speaking to other humans. I would not call you out so much, but having seen plenty of evidence of the way you operate in the closed but pending PRs, I really felt it needed addressing.